src/model/services/mapi/generic/getListResource.ts

```diff
@@ -0,0 +1,55 @@
+import { GenericResponseError } from 'model/clients/GenericResponseError';
+import { IHttpClient } from 'model/clients/HttpClient';
+import { StatusCodes } from 'model/constants';
+import ErrorReporter from 'utils/errors/ErrorReporter';
+import { IOAuth2Provider } from 'utils/OAuth2/OAuth2';
+
+import { IList } from '../metav1';
+import { getResource } from './getResource';
+
+const MAX_ATTEMPT_COUNT = 5;
+const MAX_RETRY_DELAY = 5000; // 5 seconds
+const SECOND = 1000; // 1 second
+
+function tryGetListResource<T extends IList<{}>>(
+  client: IHttpClient,
+  auth: IOAuth2Provider,
+  url: string,
+  attempt = 1
+): Promise<T> {
+  return getResource<T>(client, auth, url).then((data) => {
+    if (data.items === undefined) {
+      ErrorReporter.getInstance().notify(
+        `Invalid Response. Items are missing in the getListResource response.`,
+        { url, attempt }
+      );
```

Comment:
Good idea to report the number of attempts! What do you think about including the entire data object returned? Would it help us get more insight or would it be redundant?

Reply:
good idea, let's do it